data/core/docview.lua

```diff
@@ -456,8 +457,15 @@ function DocView:draw_line_text(line, x, y)
 end
 
 function DocView:draw_caret(x, y)
-    local lh = self:get_line_height()
+  local lh = self:get_line_height()
+  if self.doc.overwrite then
+    local line, col = self.doc:get_selection_idx(self.current_caret)
+    local charw = self:get_font():get_width(self.doc:get_char(line, col))
+    renderer.draw_rect(x, y + lh, charw, style.caret_width * 2, style.caret)
+  else
     renderer.draw_rect(x, y, style.caret_width, lh, style.caret)
+  end
+  self.current_caret = self.current_caret + 1
 end
```

Comment:
A better idea would be to either:
1. Provide a new function `DocView:draw_overtype_caret(x, y, w)`
1. Add additional parameters to `DocView:draw_caret(x, y, line, col, overtype)`

Having it rely on the order of draws is too finicky.

Reply:
idk about "draw_overtype_caret" though, also what would overtype be?

Reply:
Overtype is the common name of the feature this PR is implementing!

Reply:
no, like, if it's a boolean or what is it

Reply:
Ah yeah, it would be a boolean.

Reply:
also isn't said parameter kinda useless as I put `self.overwrite`?

Reply:
For example we wouldn't respect that when using IME.

Reply:
ah